openshift-knative-operator/pkg/serving/extension.go

```diff
@@ -201,3 +206,29 @@ func (e *extension) fetchLoggingHost(ctx context.Context) string {
 	}
 	return route.Status.Ingress[0].Host
 }
+
+func (e *extension) cleanupDomainMapping(ctx context.Context, ks *operatorv1beta1.KnativeServing) error {
+	client := e.kubeclient
+
+	for _, dep := range []string{"domain-mapping", "domainmapping-webhook"} {
+		if err := client.AppsV1().Deployments(ks.GetNamespace()).Delete(ctx, dep, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete deployment %s: %w", dep, err)
+		}
+	}
```

Comment:
Should this part be more defensive and jump out as soon as no `Deployments` are found?